Common/TableProducer/trackselection.cxx

```diff
@@ -53,32 +54,37 @@ struct TrackSelectionTask {
     switch (itsMatching) {
       case 0:
         // Run 2 SPD kAny
-        globalTracks = getGlobalTrackSelection();
-        break;
+        if (!isRun3) {
+          globalTracks = getGlobalTrackSelection();
+          break;
+        }
       case 1:
         // Run 3 kAny on 3 IB layers of ITS
         if (isRun3) {
           globalTracks = getGlobalTrackSelectionRun3ITSMatch(TrackSelection::GlobalTrackRun3ITSMatching::Run3ITSibAny);
+          break;
         }
-        break;
       case 2:
         // Run 3 kAny on all 7 layers of ITS
         if (isRun3) {
           globalTracks = getGlobalTrackSelectionRun3ITSMatch(TrackSelection::GlobalTrackRun3ITSMatching::Run3ITSallAny);
+          break;
         }
-        break;
       case 3:
         // Run 3 kAll on all 7 layers of ITS
         if (isRun3) {
           globalTracks = getGlobalTrackSelectionRun3ITSMatch(TrackSelection::GlobalTrackRun3ITSMatching::Run3ITSall7Layers);
+          break;
         }
-        break;
       default:
         LOG(fatal) << "TrackSelectionTask with undefined cuts. Fix it!";
         break;
     }
     globalTracks.SetPtRange(ptMin, ptMax);
     globalTracks.SetEtaRange(etaMin, etaMax);
+    if (compatibilityIU) {
```

Comment:
why is this needed? to run over old AO2Ds (like OCT pilot beam pass1,2,3) where no `tracksIU` were present?

Reply:
correct, now all tracks are discarded e.g. in LHC22c5